mlflow/utils/validation.py

```diff
@@ -96,6 +133,47 @@ def _validate_experiment_id(exp_id):
                               error_code=INVALID_PARAMETER_VALUE)
 
 
+def _validate_batch_limit(entity_name, limit, length):
+    if length > limit:
+        error_msg = ("A batch logging request can contain at most {limit} {name}. "
+                     "Got {count} {name}. Please split up {name} across multiple requests and try "
+                     "again.").format(name=entity_name, count=length, limit=limit)
+        raise MlflowException(error_msg, error_code=INVALID_PARAMETER_VALUE)
+
+
+def _validate_batch_log_limits(metrics, params, tags):
+    """Validate that the provided batched logging arguments are within expected limits."""
+    _validate_batch_limit(entity_name="metrics", limit=MAX_METRICS_PER_BATCH, length=len(metrics))
+    _validate_batch_limit(entity_name="params", limit=MAX_PARAMS_TAGS_PER_BATCH, length=len(params))
+    _validate_batch_limit(entity_name="tags", limit=MAX_PARAMS_TAGS_PER_BATCH, length=len(tags))
+    total_length = len(metrics) + len(params) + len(tags)
+    _validate_batch_limit(entity_name="metrics, params, and tags",
+                          limit=MAX_ENTITIES_PER_BATCH, length=total_length)
+
+
+def _validate_batch_log_data(metrics, params, tags):
+    for metric in metrics:
+        _validate_metric(metric.key, metric.value, metric.timestamp)
+        # TODO: move _validate_length_limit calls into _validate_metric etc. This would be a
```

Comment:
👍 

This comment was exactly where my eyes wandered around for this anomalous check! WTG..